editor/src/components/inspector/sections/component-section/data-picker-popup.tsx

```diff
@@ -93,91 +95,190 @@ export const DataPickerPopup = React.memo(
+  const colorTheme = useColorTheme()
+  const [selectedIndex, setSelectedIndex] = React.useState<number>(0)
+  const totalChildCount = variableOption.variableChildren?.length ?? 0
+
+  const {
+    variableName,
+    definedElsewhere,
+    value,
+    displayName,
+    depth = 0,
+    variableChildren,
+  } = variableOption
+  const isArray = variableOption.variableType === 'array'
+  const tweakProperty = onTweakProperty(variableName, definedElsewhere)
+  const stopPropagation = useCallback((e: React.MouseEvent) => {
+    e.stopPropagation()
+  }, [])
+  return (
+    <>
+      <Button
+        data-testid={VariableFromScopeOptionTestId(idx)}
+        key={variableName}
+        style={{ width: '100%', height: 25 }}
+        onClick={stopPropagation}
+      >
+        <UIGridRow
+          padded={false}
+          variant='<--1fr--><--1fr-->'
+          style={{
+            justifyContent: 'space-between',
+            alignItems: 'flex-start',
+            gap: 8,
+            width: '100%',
+            minHeight: 'auto',
+            gridTemplateColumns: '48% 48%',
+          }}
+        >
+          <div onClick={tweakProperty}>
+            <span
+              style={{
+                marginLeft: 4 * depth,
+                borderRadius: 2,
+                fontWeight: 400,
+                display: 'flex',
+                maxWidth: '100%',
+              }}
+            >
+              {depth > 0 ? (
+                <span
+                  style={{
+                    borderLeft: `1px solid ${colorTheme.neutralBorder.value}`,
+                    borderBottom: `1px solid ${colorTheme.neutralBorder.value}`,
+                    width: 5,
+                    display: 'inline-block',
+                    height: 10,
+                    marginRight: 4,
+                    position: 'relative',
+                    top: 0,
+                    marginLeft: (depth - 1) * 8,
+                    flex: 'none',
+                    textOverflow: 'ellipsis',
+                    overflow: 'hidden',
+                  }}
+                ></span>
+              ) : null}
+              <span
+                style={{
+                  textOverflow: 'ellipsis',
+                  overflow: 'hidden',
+                }}
+              >
+                {displayName}
+              </span>
+            </span>
+          </div>
+          <div
+            style={{
+              display: 'flex',
+              justifyContent: 'flex-end',
+              width: '94%',
+            }}
+            onClick={isArray ? stopPropagation : tweakProperty}
+          >
+            <span
+              style={{
+                fontWeight: 400,
+                color: colorTheme.neutralForeground.value,
+                textOverflow: 'ellipsis',
+                maxWidth: 130,
+                overflow: 'hidden',
+              }}
+            >
+              {isArray ? (
+                <Selector
```

Comment:
ultra nitpick, but selector is not the most straightforward name! could be IndexSelector or ArrayPaginator